src/widgets/modals/ModalContainer.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { StyleSheet, Text, TouchableOpacity, View, Modal } from 'react-native';
+
+import { CloseIcon } from '../icons';
+
+import {
+  APP_FONT_FAMILY,
+  FULL_SCREEN_MODAL_MARGIN,
+  PAGE_CONTENT_PADDING_HORIZONTAL,
+  DARKER_GREY,
+} from '../../globalStyles';
+
+/**
+ * A modal that can be displayed over the page content container, rendering any children
+ * about two thirds of the way up, and a cross in the top right to close
+ * @prop {Bool}             fullScreen Force the modal to cover the entire screen.
+ * @prop {Bool}             isVisible  Whether the modal is open
+ * @prop {Func}             onClose    A function to call if the close x is pressed
+ * @prop {String}           title      The title to show in within the modal.
+ * @prop {React.Element}    children   The components to render within the modal
+ */
+const ModalContainer = ({ fullScreen, isVisible, onClose, title, children, noCancel }) => {
+  const {
+    contentContainer,
+    modalContainer,
+    titleFont,
+    titleBar,
+    closeButtonContainer,
+    childrenContainer,
+    fullScreenChildrenContainer,
+    flexSpacer,
+    closeButton,
+    fullScreenContentContainer,
+  } = localStyles;
+
+  const internalChildrenContainer = fullScreen ? fullScreenChildrenContainer : childrenContainer;
+  const internalContentContainer = fullScreen ? fullScreenContentContainer : contentContainer;
+
+  const CloseButton = () => (
+    <TouchableOpacity onPress={onClose} style={closeButton}>
+      <CloseIcon />
+    </TouchableOpacity>
+  );
+
+  const TitleBar = () => (
+    <View style={titleBar}>
+      <View style={flexSpacer} />
+      {!!title && <Text style={titleFont}>{title}</Text>}
+      {onClose && !noCancel && (
+        <View style={closeButtonContainer}>
+          <CloseButton />
+        </View>
+      )}
+    </View>
+  );
+
+  return (
+    <View style={modalContainer}>
+      <Modal
+        visible={isVisible}
+        animationType="slide"
+        fullScreen={fullScreen}
+        transparent={true}
```

Comment:
I'm ok with doing boolean shortcut:
```suggestion
        transparent
```